src/assets/js/fullpostpage/unlock-content/edit-fullpp.js

```diff
@@ -0,0 +1,302 @@
+import { __ } from "@wordpress/i18n";
+import {
+  PanelBody,
+  PanelRow,
+  TextControl,
+  SelectControl,
+  Button,
+  Notice,
+} from "@wordpress/components";
+import { useEffect, useState } from "react";
+import apiFetch from "@wordpress/api-fetch";
+import { useSelect } from '@wordpress/data';
+import "../../../scss/admin/editor.scss";
+
+const EditFullPP = ({ locks, ethereumNetworks, onUpdateLocks, onUpdateEthereumNetworks }) => {
+  
+  const [saveMessage, setSaveMessage] = useState(null);
+  const [refresh, setRefresh] = useState(false); //refresh lock(s) meta panel to keep track of locks already added
+
+
+  //get current post id
+  const getpost = useSelect((select) => {
+    const { getCurrentPostId } = select("core/editor");
+    const id = getCurrentPostId();
+  
+    return {
+      id,
+    };
+  }, []);
+
+  //check if current post is already 
+  const { isSavingPost, didPostSaveRequestSucceed } = useSelect((select) => {
+    const { isSavingPost, didPostSaveRequestSucceed } = select("core/editor");
+    return {
+      isSavingPost: isSavingPost(),
+      didPostSaveRequestSucceed: didPostSaveRequestSucceed(),
+    };
+  }, []);
+
+
+  //fetch locks attributes
+  const fetchLockSettings = async (post_id) => {
+    try {
+      const response = await apiFetch({
+        path: `/unlock-protocol/v1/get_unlockpfullpp_attributes?post_id=${post_id}`,
+      });
+
+      if (response && response.locks) {
+        onUpdateLocks(response.locks);
+      }
+    } catch (error) {
+      console.error("Error fetching lock settings:", error);
+    }
+  };
+
+  //useEffect 1: that fetches the networks
+  useEffect(() => {
+    apiFetch({
+      path: "/unlock-protocol/v1/settings",
+    })
+      .then((resp) => {
+        let networks = resp.networks;
+        let selectOptions = [
+          {
+            label: __("None", "unlock-protocol"),
+            value: -1,
+          },
+        ];
+
+        Object.entries(networks).forEach(([key, item]) => {
+          selectOptions.push({
+            label: item.network_name,
+            value: item.network_id,
+          });
+        });
+
+        onUpdateEthereumNetworks(selectOptions);
+
+        // Fetch lock settings from the post meta after networks are fetched
+        fetchLockSettings(getpost.id);
+      })
+      .catch((err) => { });
+  }, [getpost.id, refresh]); // refresh the useEffect anytime either of this changes
```

Comment:
The networks don't change much so I think it is ok to just fetch this once and not when`getpost.id` changes

Reply:
This is not just network, it required keeping the previous lock(s) setup be persistently displayed to the user when publish/update post/page button clicked or when the editor is fully reloaded. or else the user will not be able to interact with previously set lock(s).
The implementation is as needed to work with the post/page meta options panel sidebar.
Kindly consider this resolved too.